code/data_structures/stack/balanced_expression/balanced_expression.cpp

```diff
@@ -0,0 +1,51 @@
+#include<iostream>
+using namespace std;
+
+bool isValid(char expression[]) {
+
+    char stack[100];
+    int len=0;
+    int expressionLength = strlen(expression);
+    
+    for (int i=0;i<expressionLength;i++) {
+        if (expression[i] == '(' || expression[i] == '{' || expression[i] == '[')
+            stack[len++] = expression[i];
+        else {
+            if (expression[i] == ')')
+                if (stack[len-1] == '(')
+                    len--;
+                else
+                    return false;
+            else if (expression[i] == '}')
+                if (stack[len-1] == '{')
+                    len--;
+                else
+                    return false;
+            else if (expression[i] == ']')
+                if (stack[len-1] == '[')
+                    len--;
+                else
+                    return false;
+        }
+    }
+
+    if (len==0)
+        return true;
+    else
+        return false;
```

Comment:
It could be simplified to `return (len == 0);` (or `stack.empty()` when using a `stack`)